src/converse-core.js

```diff
@@ -4893,7 +4893,12 @@
                         throw new Error("initConnection: If you use auto_login and "+
                             "authentication='login' then you also need to provide a password.");
                     }
-                    this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+Strophe.getResourceFromJid(this.jid);
+                    var resource = Strophe.getResourceFromJid(this.jid);
+                    if (!resource) {
+                        this.jid = this.jid.toLowerCase() + '/converse.js-' + Math.floor(Math.random()*139749825).toString();
+                    } else {
+                        this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+Strophe.getResourceFromJid(this.jid);
+                    }
```

Comment:
This line can be: `this.jid = Strophe.getBareJidFromJid(this.jid).toLowerCase()+'/'+resource;`